### ISSUE TITLE
allObjectsInRealm behaves strangely

### ISSUE THREAD

**Issue by tspecht** [realm/realm-cocoa#1134]
Hi guys,

sorry to bother you again but I ran into another issue today regarding the usage of allObjectsInRealm, no matter if used on the default realm or on a custom one. But this time, I'm able to shed some more light on my issue :)

My setup includes ~1500 objects stored in the default Realm which are the main dataset for my application. I wanted to add the possibility to tag them and therefore introduced a new Tag model to my application which was initially stored in a separate realm to reduce the possible occurrences of the concurrent access issues I am also still seeing. In addition, I implemented a very naiv search which looped the tags to find matching ones and afterwards fetched all the main objects associated with those tags. I am doing so in a background thread using dispatch_async in conjunction with Bolts. 

To get my initial dataset which afterwards was further narrowed down using some objectsWhere: calls and afterwards looped, I used the following statements:

RLMResults\* filteredSnips = [MyModel allObjects];
RLMResults\* filteredTags = [Tag allObjectsInRealm:[RLMRealm tagRealm]];

Surprisingly, the second statement caused a lot of problems: While in the search view, a lot of crashed trying to access the MyModel objects properties occurred, when no crash occurred and I would go back to the previous screen which is basically a list of most of the MyModel objects, some boolean properties would be totally mixed up (i.e. they all were inverted). Restarting the application fixed the problem until another search was attempted. The interesting part of the story is, that the crashes only occurred on two string columns added with the last migration, specifically in row.hpp line 319. Also, when not updating the application from the previous version but instead doing a fresh install, the problem doesn't appear at all.

Therefore I thought the problem was the underlying migration which was added to use the two string columns and simply did newObject[@"tags"] = @"".
After hours of investigation and manually commenting and uncommenting code out at my side I found a very strange solution: When using [Tag allObjects] directly and storing the tag objects also in the default realm all magically begins to work, even in the update scenario. I checked the implementation of allObject and it simply does call allObjectsInRealm:RLMRealm.defaultRealm which I would suppose to be the same as a call to allObjectsInRealm:myCustomRealm.

Anyways, my application seems to be running for now but this issue seems kinda odd to me. Is there any chance you guys can git it fixed? If you need any more info or feedback please let me know. Realm is doing an incredible work when it comes to pure performance on a single thread but seems a little unstable when used in parallel scenarios. I would like to help you improve the implementation! :)
